sqlmesh/core/model/kind.py

```diff
@@ -153,53 +159,76 @@ def data_hash_values(self) -> t.List[t.Optional[str]]:
 
 
 class TimeColumn(PydanticModel):
-    column: str
+    column: exp.Expression
     format: t.Optional[str] = None
 
     @classmethod
     def validator(cls) -> classmethod:
-        def _time_column_validator(v: t.Any) -> TimeColumn:
+        def _time_column_validator(v: t.Any, values: t.Any) -> TimeColumn:
+            values = values if isinstance(values, dict) else values.data
+            dialect = values.get("dialect")
             if isinstance(v, exp.Tuple):
-                kwargs = {
-                    key: v.expressions[i].name
-                    for i, key in enumerate(("column", "format")[: len(v.expressions)])
-                }
-                return TimeColumn(**kwargs)
-
-            if isinstance(v, exp.Expression):
-                return TimeColumn(column=v.name)
+                column_expr = v.expressions[0]
+                column = (
+                    exp.column(column_expr)
+                    if isinstance(column_expr, exp.Identifier)
+                    else column_expr
+                )
+                format = v.expressions[1].name if len(v.expressions) > 1 else None
+            elif isinstance(v, exp.Expression):
+                column = exp.column(v) if isinstance(v, exp.Identifier) else v
```

Comment:
can you just make this a validator for TimeColumn

Reply:
No, because I'll also have to propagate dialect into `TimeColumn` and not just `kind`.